prql/src/ast.rs

```diff
@@ -136,9 +127,50 @@ pub struct NamedArg {
 }
 
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-pub struct Assign {
-    pub lvalue: Ident,
-    pub rvalue: Box<Item>,
+pub struct NamedExpr {
+    pub alias: Option<Ident>,
+    pub expr: Box<Item>,
+}
+
+impl NamedExpr {
+    pub fn unnamed(expr: Item) -> Self {
+        NamedExpr {
+            alias: None,
+            expr: Box::new(expr),
+        }
+    }
+
+    /// Almost all function parameters and list items are wrapped into NamedExpr, which
+    /// can be passed either to this function or to into_list_with_names.
+    pub fn discard_name(self) -> Result<Item> {
+        if self.alias.is_some() {
+            bail!("Cannot use alias for: {self:?}")
+        }
+        Ok(*self.expr)
+    }
+
+    pub fn map<F>(self, mut f: F) -> Self
```

Comment:
This is reasonable; FYI this is somewhat what the `AstFold` is for — letting this sort of operation run over any type (it could be called `AstMap` tbh).

Reply:
Oh, yes this can go - it was here for some other reason and it's not needed any more.